src/java.base/share/classes/java/lang/constant/DirectMethodHandleDescImpl.java

```diff
@@ -63,9 +63,10 @@ final class DirectMethodHandleDescImpl implements DirectMethodHandleDesc {
      * @jvms 4.2.2 Unqualified Names
      */
     DirectMethodHandleDescImpl(Kind kind, ClassDesc owner, String name, MethodTypeDesc type) {
-        if (kind == CONSTRUCTOR)
-            name = "<init>";
-
+        if (kind == CONSTRUCTOR) {
+            // TODO: missing "value objects", they don't use Q-type desc
```

Comment:
Needs fixing to support Value Objects (currently only correct for primitive value types)

Reply:
My patch I sent to you already fixed this issue.